realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -70,6 +73,16 @@ static RealmCacheType valueOf(Class<? extends BaseRealm> clazz) {
     // Separated references and counters for typed Realm and dynamic Realm.
     private final EnumMap<RealmCacheType, RefAndCount> refAndCountMap;
 
+    /**
+     * This is a global reference of all the Realm/DynamicRealm instances across thread. As {@link #refAndCountMap}
+     * allows its {@link #refAndCountMap#localRealm} to yield {@link BaseRealm} instance to a specific thread,
+     * it is necessary to have a global list of BaseRealm that could be approached from any thread.
+     *
+     * <p>As one can see there is no protection on this list from being accessed via multiple threads.
+     * Please take caution when this is exposed to un-synchronized methods.
+     */
+    private final ArrayList<WeakReference<BaseRealm>> globalBaseRealmReference;
```

Comment:
I don't think we need this. See e.g. how `Realm.deleteRealm()` access the global count?